### DISCUSSION TITLE
Can't deploy my pages<title>

### DISCUSSION THREAD

**Discussion by salmakhaled74** [community/community#47992] [Actions]
### Select Topic Area

Product Feedback

### Body
every check-in passes but there is a 404 error
https://github.com/salmakhaled74/salmakhaled74.github.io

**Comment by airtower-luna** [comment]
There's no `index.html` file at the top of your repository, that's why accessing `https://salmakhaled74.github.io/` returns 404. The individual files are deployed, e.g. https://salmakhaled74.github.io/package.json works.

I assume the `public/index.html` file is what you want to be at the top level. You could either move it (make sure that the paths still work) or deploy only the `public/` directory (assuming it contains all that should be deployed).

**Comment by salmakhaled74** [comment]
What about server.js it's outside public directory

**Comment by airtower-luna** [comment]
The `server.js` isn't included by the `index.html` file. I guess it's a Node.js service? That won't work on GitHub Pages at all, only static sites are supported.

**Comment by salmakhaled74** [comment]
So what do you recommend? I'm still a beginner and have not much experience at this

**Comment by airtower-luna** [comment]
If possible (I haven't looked deeply at your code) you can move the code that currently runs on the server into the client side, so you don't need the server-side application. Then you can just deploy the client and be done.

If that's not possible and you need (or want) the server-side application, you'll need to find a different place to host your site, one that lets you run things on the server.
GitHub Pages allows only static sites, no server-side applications.

**Comment by salmakhaled74** [comment]
Thank you so much for your help I will search for another website
